overviewer_core/src/primitives/overlay-heatmap.c

```diff
@@ -0,0 +1,93 @@
+/*
+ * This file is part of the Minecraft Overviewer.
+ *
+ * Minecraft Overviewer is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or (at
+ * your option) any later version.
+ *
+ * Minecraft Overviewer is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with the Overviewer.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <math.h>
+#include "overlay.h"
+
+typedef struct {
+    /* inherits from overlay */
+    RenderPrimitiveOverlay parent;
+    int t_invisible;
+    int delta_t;
+} RenderPrimitiveHeatmap;
+
+static void get_color(void* data, RenderState* state,
+                      uint8_t* r, uint8_t* g, uint8_t* b, uint8_t* a) {
+    RenderPrimitiveHeatmap* self = (RenderPrimitiveHeatmap*)data;
+    long mtime;
+    float _value_f;
+    char value;
+    PyObject *mtime_pyobj;
+
+    // Set default values (will get overridden based on self->mode)
+    *r = 255;
+    *g = 0;
+    *b = 0;
+
+    // Get the chunk modified time
+    mtime_pyobj = PyObject_CallMethod(state->regionset, "get_chunk_mtime", "ii", state->chunkx, state->chunkz);
+    if (mtime_pyobj == NULL) {
```

Comment:
Should check against `Py_None`

Reply:
Thanks for the Info, yes that should indeed check for `Py_None` according to the docs  `PyObject_CallMethod`returns NULL on errors or exceptions. Since it probably doesn't hurt I have changed it to check for `NULL` and `PY_None`, catching exceptions in python should any occur and checking for None as return value.

Reply:
I highly suspect Py_None is simply an alias to NULL, I think checking for NULL is fine, in retrospect.